django/contrib/sessions/middleware.py

```diff
@@ -57,10 +57,11 @@ def process_response(self, request, response):
                         try:
                             request.session.save()
                         except UpdateError:
-                            # The user is now logged out; redirecting to same
-                            # page will result in a redirect to the login page
-                            # if required.
-                            return redirect(request.path)
+                            # Session has been deleted (the user is logged out or so).
+                            # Raise SuspiciousOperation exception indicating that
+                            # request can't be properly performed.
+                            msg = 'Session had been destroyed before request completed (user logged out and etc.)'
```

Comment:
Is this wording accurate and better? "The request's session was deleted before the request completed. The user may have logged out in a concurrent request, for example." I think the comments could be deleted in that case.
Use this style:
```
raise SuspiciousOperation(
    "..."
    "..."
)
```